logbus/run.go

```diff
@@ -11,21 +11,23 @@ import (
 
 // Run is a run logstream delta generator for a run.
 type Run struct {
-	b        *Bus
-	mu       sync.Mutex
-	targets  map[string]*Target
-	commands map[string]*Command
-	ended    bool
+	b          *Bus
+	mu         sync.Mutex
+	targets    map[string]*Target
+	commands   map[string]*Command
+	ended      bool
+	mainTarget bool
 
 	generic *Generic
 }
 
 func newRun(b *Bus) *Run {
 	run := &Run{
-		b:        b,
-		targets:  make(map[string]*Target),
-		commands: make(map[string]*Command),
-		generic:  nil, // set below
+		b:          b,
+		targets:    make(map[string]*Target),
+		commands:   make(map[string]*Command),
+		generic:    nil, // set below
+		mainTarget: true,
```

Comment:
nit:
For clarity, we should rename this to `hasMainTarget` as that seems to be the purpose.
(If we did change it, we'd have to negate the boolean).